### DISCUSSION TITLE
grain cannot be called but has data

### DISCUSSION THREAD

**Discussion by Remsdel** [mtex-toolbox/mtex#1324] [Ask Anything]
Hello,
Probably a trivial problem but I am really stuck with this one:
I do a classical grain reconstruction 
[grains,ebsd.grainId] = calcGrains(ebsd('indexed'),'angle',5*degree,'boundary','tight');
I performed my code on many EBSD dataset without issues but there is one map where one grain gives me an error message when I call it directly:

          **grains(1360)**
          Unable to perform assignment because the indices on the left side are not compatible with
          the size of the right side.
          
          Error in grain2d/subSet (line 68)
            inverseorder(ind2) = ind1;
          
          Error in grain2d/subsref (line 26)
            grains = subSet(grains,ind);

The same grain seems however to have data in all the sub-category: grains.id, grains.phase, grains.id2ind, grains.GOS etc.. so it really puzzling to me why I get the error message when calling the grain directly (which then become an issue for further processing of data)
I do not do any other operation after the grain reconstruction.
Any help welcome,

Cordially,

Rémi Delville

**Comment by kilir** [comment]
Hi,

what happens when you address it by id? 

Cheers,
Rüdiger

**Comment by Remsdel** [comment]
Hi,
Yes this is the strange part about it, I can address the grain by any of the sub categories, i.e. grains.id(1360) but when calling it directly grains(1360) I get the error message. I got around it by modifying my grain reconstruction parameters (using no tight boundary) but this remains a rather strange issue.
Cordially,
Rémi

**Reply by kilir** [reply]
I meant `grains('id',grains.id(1360) )`.

If I remember correctly, this has been reported here in the forum also for this very single case some while ago. Changing the default parameter (0.95) associated with tight has helped. E.g.
`calcgrains(ebsd,'thight',0.85)`

Cheers,
Rüdiger
